### PR TITLE
allow extending of trusted hosts

### PR DESCRIPTION
**Pull request by teleaziz**

### CODE REVIEW

File: packages/core/src/builder.class.ts [hunk 1]
```diff
@@ -1263,15 +1271,11 @@ export class Builder {
         const url = parse(event.origin);
         const isRestricted =
           ['builder.register', 'builder.registerComponent'].indexOf(event.data?.type) === -1;
-        if (
-          isRestricted &&
-          !(
-            url.hostname &&
-            (url.hostname === 'builder.io' ||
-              url.hostname.endsWith('.builder.io') ||
```
Comment (steve8708) [inline packages/core/src/builder.class.ts:1271]: this update negates this piece of code - we want to support **.builder.io, not just builder.io

File: packages/core/src/builder.class.ts [hunk 2]
```diff
@@ -422,6 +422,7 @@ export class Builder {
   authToken = '';
 
   static editors: any[] = [];
+  static trustedHosts: string[] = ['builder.io', 'localhost:1234'];
```
Comment (steve8708) [inline packages/core/src/builder.class.ts:425]:
```suggestion
  static trustedHosts: string[] = ['builder.io', 'localhost'];
```
this check is against `hostname` which excludes the port (ant not `host` which includes it)

File: packages/core/src/builder.class.ts (hunk 1 above)
Comment (teleaziz) [reply packages/core/src/builder.class.ts:1271]: updated to use `isTrusted(host)` which only check if the `builder.io` is in the host string

File: packages/core/src/builder.class.ts [hunk 3]
```diff
@@ -489,6 +491,14 @@ export class Builder {
     this.actions.push(action);
   }
 
+  static registerTrustedHost(host: string) {
+    this.trustedHosts.push(host);
+  }
+
+  static isTrusted(host: string) {
```
Comment (steve8708) [inline packages/core/src/builder.class.ts:498]:
```suggestion
  static isTrustedHost(host: string) {
```

nit pick, just a little more consistent and clear imo

File: packages/core/src/builder.class.ts (hunk 1 above)
Comment (steve8708) [reply packages/core/src/builder.class.ts:1271]: 👍

File: packages/core/src/builder.class.ts [hunk 4]
```diff
@@ -489,6 +491,14 @@ export class Builder {
...
+    return this.trustedHosts.findIndex(allowedHost => host.indexOf(allowedHost) > -1) > -1;
```
Comment (steve8708) [inline packages/core/src/builder.class.ts:499]: theres still an issue here, someone could register

maliciousbuilder.io and it works

or builder.io.malicious.com

it's important for host matching that the host is either exact, or `*.${host}`

Comment (teleaziz) [reply packages/core/src/builder.class.ts:499]: make sense, I'll update 👍

**Review by steve8708 — APPROVED** [review]

**Pull request merged**